### ISSUE TITLE
[Bug]: gpu-e2e run failed reporting 'GPU_IVF_FLAT' is an invalid index type and raised exception

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#22913]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 9297ea3
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):   rocksmq 
- SDK version(e.g. pymilvus v2.0.0rc2):2.3.0.dev48
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

gpu-e2e run failed reporting 'GPU_IVF_FLAT' is an invalid index type and raised exception.
```
[2023-03-21T12:48:07.899Z] _ TestCollectionSearch.test_search_after_different_index_with_params_gpu[32-False-True-GPU_IVF_FLAT-params1] _

[2023-03-21T12:48:07.899Z] [gw2] linux -- Python 3.8.16 /usr/local/bin/python3

[2023-03-21T12:48:07.899Z] [gw2] linux -- Python 3.8.16 /usr/local/bin/python3[gw2] linux -- Python 3.8.16 /usr/local/bin/python3

[2023-03-21T12:48:07.899Z] 

[2023-03-21T12:48:07.899Z] self = <test_search.TestCollectionSearch object at 0x7f850ecf7670>, dim = 32

[2023-03-21T12:48:07.899Z] index = 'GPU_IVF_FLAT', params = {'nlist': 128}, auto_id = False, _async = True

[2023-03-21T12:48:07.899Z] 

[2023-03-21T12:48:07.899Z]     @pytest.mark.tags(CaseLabel.GPU)

[2023-03-21T12:48:07.899Z]     @pytest.mark.parametrize("index, params",

[2023-03-21T12:48:07.899Z]                              zip(ct.all_index_types[9:14],

[2023-03-21T12:48:07.899Z]                                  ct.default_index_params[9:14]))

[2023-03-21T12:48:07.899Z]     def test_search_after_different_index_with_params_gpu(self, dim, index, params, auto_id, _async):

[2023-03-21T12:48:07.899Z]         """

[2023-03-21T12:48:07.899Z]         target: test search after different index

[2023-03-21T12:48:07.899Z]         method: test search after different index and corresponding search params

[2023-03-21T12:48:07.899Z]         expected: search successfully with limit(topK)

[2023-03-21T12:48:07.899Z]         """

[2023-03-21T12:48:07.899Z]         # 1. initialize with data

[2023-03-21T12:48:07.899Z]         collection_w, _, _, insert_ids, time_stamp = self.init_collection_general(prefix, True, 5000,

[2023-03-21T12:48:07.899Z]                                                                                   partition_num=1,

[2023-03-21T12:48:07.899Z]                                                                                   auto_id=auto_id,

[2023-03-21T12:48:07.899Z]                                                                                   dim=dim, is_index=False)[0:5]

[2023-03-21T12:48:07.899Z]         # 2. create index and load

[2023-03-21T12:48:07.899Z]         if params.get("m"):

[2023-03-21T12:48:07.899Z]             if (dim % params["m"]) != 0:

[2023-03-21T12:48:07.899Z]                 params["m"] = dim // 4

[2023-03-21T12:48:07.899Z]         if params.get("PQM"):

[2023-03-21T12:48:07.899Z]             if (dim % params["PQM"]) != 0:

[2023-03-21T12:48:07.899Z]                 params["PQM"] = dim // 4

[2023-03-21T12:48:07.899Z]         default_index = {"index_type": index, "params": params, "metric_type": "L2"}

[2023-03-21T12:48:07.899Z]         collection_w.create_index("float_vector", default_index)

[2023-03-21T12:48:07.899Z]         collection_w.load()

[2023-03-21T12:48:07.899Z]         # 3. search

[2023-03-21T12:48:07.899Z] >       search_params = cf.gen_search_param(index)

[2023-03-21T12:48:07.899Z] 

[2023-03-21T12:48:07.899Z] testcases/test_search.py:1825: 

[2023-03-21T12:48:07.899Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-03-21T12:48:07.899Z] 

[2023-03-21T12:48:07.899Z] index_type = 'GPU_IVF_FLAT', metric_type = 'L2'

[2023-03-21T12:48:07.899Z] 

[2023-03-21T12:48:07.899Z]     def gen_search_param(index_type, metric_type="L2"):

[2023-03-21T12:48:07.899Z]         search_params = []

[2023-03-21T12:48:07.899Z]         if index_type in ["FLAT", "IVF_FLAT", "IVF_SQ8", "IVF_PQ"]:

[2023-03-21T12:48:07.899Z]             for nprobe in [64, 128]:

[2023-03-21T12:48:07.899Z]                 ivf_search_params = {"metric_type": metric_type, "params": {"nprobe": nprobe}}

[2023-03-21T12:48:07.899Z]                 search_params.append(ivf_search_params)

[2023-03-21T12:48:07.899Z]         elif index_type in ["BIN_FLAT", "BIN_IVF_FLAT"]:

[2023-03-21T12:48:07.899Z]             if metric_type not in ct.binary_metrics:

[2023-03-21T12:48:07.899Z]                 log.error("Metric type error: binary index only supports distance type in (%s)" % ct.binary_metrics)

[2023-03-21T12:48:07.899Z]                 # default metric type for binary index

[2023-03-21T12:48:07.899Z]                 metric_type = "JACCARD"

[2023-03-21T12:48:07.899Z]             for nprobe in [64, 128]:

[2023-03-21T12:48:07.899Z]                 binary_search_params = {"metric_type": metric_type, "params": {"nprobe": nprobe}}

[2023-03-21T12:48:07.900Z]                 search_params.append(binary_search_params)

[2023-03-21T12:48:07.900Z]         elif index_type in ["HNSW"]:

[2023-03-21T12:48:07.900Z]             for ef in [64, 32768]:

[2023-03-21T12:48:07.900Z]                 hnsw_search_param = {"metric_type": metric_type, "params": {"ef": ef}}

[2023-03-21T12:48:07.900Z]                 search_params.append(hnsw_search_param)

[2023-03-21T12:48:07.900Z]         elif index_type == "ANNOY":

[2023-03-21T12:48:07.900Z]             for search_k in [1000, 5000]:

[2023-03-21T12:48:07.900Z]                 annoy_search_param = {"metric_type": metric_type, "params": {"search_k": search_k}}

[2023-03-21T12:48:07.900Z]                 search_params.append(annoy_search_param)

[2023-03-21T12:48:07.900Z]         elif index_type == "DISKANN":

[2023-03-21T12:48:07.900Z]             for search_list in [20, 30]:

[2023-03-21T12:48:07.900Z]                 diskann_search_param = {"metric_type": metric_type, "params": {"search_list": search_list}}

[2023-03-21T12:48:07.900Z]                 search_params.append(diskann_search_param)

[2023-03-21T12:48:07.900Z]         else:

[2023-03-21T12:48:07.900Z]             log.error("Invalid index_type.")

[2023-03-21T12:48:07.900Z] >           raise Exception("Invalid index_type.")

[2023-03-21T12:48:07.900Z] E           Exception: Invalid index_type.
```

### Expected Behavior

pass

### Milvus Log

https://jenkins.milvus.io:18080/blue/organizations/jenkins/milvus-gpu-ci/detail/PR-22901/2/pipeline/110

**Comment by binbinlv** [comment]
This will be fixed by pr #22900

**Comment by binbinlv** [comment]
Fixed and close this issue.
